### PR TITLE
Add convienence constructor for Gap

### PR DESCRIPTION
**Pull request by ssteinbach**
Addresses #226

### CODE REVIEW

File: opentimelineio/schema/gap.py [hunk 1]
```diff
@@ -32,6 +35,36 @@ class Gap(core.Item):
     _serializable_label = "Gap.1"
     _class_path = "schema.Gap"
 
+    def __init__(
+        self,
+        name=None,
+        # note - only one of the following is accepted
```
Comment (avrata) [inline opentimelineio/schema/gap.py]: Super super minor, but perhaps add "two" ("only one of the following two.."). At first read I thought this applied to effects/markers/etc as well.

Comment (jminor) [reply opentimelineio/schema/gap.py]: I agree.

File: opentimelineio/schema/gap.py [hunk 2]
```diff
@@ -32,6 +35,36 @@ class Gap(core.Item):
...
+        duration=None,
+        source_range=None,
+        effects=None,
+        markers=None,
+        metadata=None,
+    ):
+        if duration and source_range:
+            raise RuntimeError(
+                "Cannot instantiate with both a source range and a duration."
+            )
```
Comment (jminor) [inline opentimelineio/schema/gap.py]: We should also check for neither and complain.

Comment (ssteinbach) [reply opentimelineio/schema/gap.py]: Unfortunately, the deserializer requires that we have the ability to `__init__` classes without any arguments, so we can't make this an error case.

**Pull request merged**